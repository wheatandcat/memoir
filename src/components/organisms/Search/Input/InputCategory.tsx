import React, { memo } from 'react';
import { StyleSheet } from 'react-native';
import View from 'components/atoms/View';
import setting from 'components/atoms/Category/setting';
import CategoryButton from 'components/molecules/CategoryButton';

export type Props = {};

const InputCategory: React.FC<Props> = () => {
  const category1 = setting().main[0];
  const category2 = setting().main[1];
  const category3 = setting().main[2];

  return (
    <>
      <View style={styles.category}>
        {category1.map((v) => (
          <View key={v} mx={2}>
            <CategoryButton
              categoryID={v}
              selected={false}
              opacity
              onPress={() => null}
              displayName={false}
            />
          </View>
        ))}
      </View>
      <View style={styles.category}>
        {category2.map((v) => (
          <View key={v} mx={2}>
            <CategoryButton
              categoryID={v}
              selected={false}
              opacity
              onPress={() => null}
              displayName={false}
            />
          </View>
        ))}
      </View>
      <View style={styles.category}>
        {category3.map((v) => (
          <View key={v} mx={2}>
            <CategoryButton
              categoryID={v}
              selected={false}
              opacity
              onPress={() => null}
              displayName={false}
            />
          </View>
        ))}
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  category: {
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
});

export default memo(InputCategory);
