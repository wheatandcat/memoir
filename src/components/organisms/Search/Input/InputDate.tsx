import React, { memo, useState } from 'react';
import { StyleSheet, TouchableOpacity, ViewStyle } from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';
import DateTimePickerModal from 'react-native-modal-datetime-picker';
import dayjs from 'lib/dayjs';

export type Props = {
  startDate: Date;
  endDate: Date;
  error?: boolean;
  onChangeStartDate: (date: Date) => void;
  onChangeEndDate: (date: Date) => void;
};

const InputDate: React.FC<Props> = (props) => {
  const [openStartDate, setOpenStartDate] = useState(false);
  const [openEndDate, setOpenEndDate] = useState(false);

  const dateText: ViewStyle[] = [styles.dateText];

  if (props.error) {
    dateText.push(styles.errorDateText);
  }

  return (
    <View pt={3} style={styles.inputDate}>
      <View style={dateText}>
        <TouchableOpacity onPress={() => setOpenStartDate(true)}>
          <Text variants="middle" color={props.error ? 'error' : 'secondary'}>
            {dayjs(props.startDate).format('YYYY.MM.DD')}
          </Text>
        </TouchableOpacity>
        <DateTimePickerModal
          isVisible={openStartDate}
          mode="date"
          locale="ja"
          confirmTextIOS="完了"
          cancelTextIOS="キャンセル"
          is24Hour={true}
          date={props.startDate}
          onConfirm={(v) => {
            setOpenStartDate(false);
            props.onChangeStartDate(v);
          }}
          onCancel={() => setOpenStartDate(false)}
        />
      </View>
      <View mx={2}>
        <Text color={props.error ? 'error' : 'secondary'}>-</Text>
      </View>
      <View style={dateText}>
        <TouchableOpacity onPress={() => setOpenEndDate(true)}>
          <Text variants="middle" color={props.error ? 'error' : 'secondary'}>
            {dayjs(props.endDate).format('YYYY.MM.DD')}
          </Text>
        </TouchableOpacity>
        <DateTimePickerModal
          isVisible={openEndDate}
          mode="date"
          locale="ja"
          confirmTextIOS="完了"
          cancelTextIOS="キャンセル"
          is24Hour={true}
          date={props.endDate}
          onConfirm={(v) => {
            setOpenEndDate(false);
            props.onChangeEndDate(v);
          }}
          onCancel={() => setOpenEndDate(false)}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  inputDate: {
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
  dateText: {
    borderColor: theme().color.secondary.main,
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
  errorDateText: {
    borderColor: theme().color.error.main,
    borderBottomWidth: 1,
  },
});

export default memo(InputDate);
