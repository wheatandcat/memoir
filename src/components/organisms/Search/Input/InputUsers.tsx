import React, { memo } from 'react';
import { StyleSheet } from 'react-native';
import View from 'components/atoms/View';
import { User } from 'queries/api/index';
import UserButton, {
  Props as UserButtonProps,
} from 'components/molecules/User/Button';

export type Props = {
  users: Pick<User, 'id' | 'image'>[];
  userIDList: string[];
} & Pick<UserButtonProps, 'onAdd' | 'onRemove'>;

const InputUsers: React.FC<Props> = (props) => {
  return (
    <View style={styles.users}>
      {props.users.map((v) => (
        <View px={3} key={v.id}>
          <UserButton
            user={v}
            size={50}
            selected={!!props.userIDList.find((v1) => v1 === v.id)}
            onAdd={props.onAdd}
            onRemove={props.onRemove}
          />
        </View>
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  users: {
    justifyContent: 'center',
    flexWrap: 'wrap',
    alignItems: 'center',
    flexDirection: 'row',
    width: '80%',
  },
});

export default memo(InputUsers);
