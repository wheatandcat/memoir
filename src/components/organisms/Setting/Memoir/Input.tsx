import React, { memo, useState, useCallback } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';
import RNPickerSelect from 'react-native-picker-select';
import DateTimePickerModal from 'react-native-modal-datetime-picker';
import dayjs from 'lib/dayjs';

export type Props = {
  dayOfWeek: number;
  time: Date;
  onChangeDayOfWeek: (dayOfWeek: number) => void;
  onChangeTime: (time: Date) => void;
};

const dayOfWeekItems = [
  { label: '日', value: 1, key: 1 },
  { label: '月', value: 2, key: 2 },
  { label: '火', value: 3, key: 3 },
  { label: '水', value: 4, key: 4 },
  { label: '木', value: 5, key: 5 },
  { label: '金', value: 6, key: 6 },
  { label: '土', value: 7, key: 7 },
];

const Input: React.FC<Props> = (props) => {
  const [openTime, setOpenTime] = useState(false);

  const onChangeTime = useCallback(
    (val: Date) => {
      setOpenTime(false);
      props.onChangeTime(val);
    },
    [props]
  );

  return (
    <View pt={4}>
      <View style={styles.input}>
        <View style={styles.inputItem}>
          <RNPickerSelect
            fixAndroidTouchableBug={true}
            useNativeAndroidPickerStyle={false}
            placeholder={{}}
            value={props.dayOfWeek}
            onValueChange={props.onChangeDayOfWeek}
            items={dayOfWeekItems}
            style={{
              placeholder: {
                width: 40,
                height: 50,
                color: theme().color.transparent,
              },
              inputIOS: {
                fontSize: theme().fontSizes.xl,
                fontWeight: theme().fontWeights.bold,
                color: theme().color.secondary.main,
                width: 40,
                height: 50,
                paddingTop: theme().space(3),
              },
              inputAndroid: {
                fontSize: theme().fontSizes.xl,
                fontWeight: theme().fontWeights.bold,
                color: theme().color.secondary.main,
                width: 40,
                height: 50,
                marginTop: theme().space(2),
                paddingTop: theme().space(1),
              },
            }}
          />
        </View>
        <View px={2} pt={4}>
          <Text>曜日の</Text>
        </View>
        <View pt={2} mt={1}>
          <TouchableOpacity>
            <Text
              fontWeight="bold"
              variants="large"
              style={styles.numberInput}
              onPress={() => setOpenTime(true)}
            >
              {dayjs(props.time).format('HH:mm')}
            </Text>
          </TouchableOpacity>
          <DateTimePickerModal
            isVisible={openTime}
            mode="time"
            locale="ja"
            confirmTextIOS="完了"
            cancelTextIOS="キャンセル"
            is24Hour={true}
            date={props.time}
            onConfirm={onChangeTime}
            onCancel={() => setOpenTime(false)}
          />
        </View>
        <View px={2} pt={4}>
          <Text>から</Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  input: {
    justifyContent: 'center',
    flexDirection: 'row',
  },
  inputItem: {},
  numberInput: {},
});

export default memo(Input);
