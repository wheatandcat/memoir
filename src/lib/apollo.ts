import { setContext } from '@apollo/client/link/context';
import { Alert } from 'react-native';
import { onError } from '@apollo/client/link/error';
import { ApolloClient, InMemoryCache, createHttpLink } from '@apollo/client';
import { storageKey, getItem } from 'lib/storage';
import Auth from 'lib/auth';

export const cache = new InMemoryCache();
const auth = new Auth();

type Param = {
  Authorization?: string;
  UserID?: string;
};

const makeApolloClient = async () => {
  const uri = `${process.env.API_HOST}/query`;

  const authLink = setContext(async (_, { headers }) => {
    const h = headers;
    const param: Param = {};
    const token = await auth.getIdToken();

    if (token) {
      param.Authorization = `Bearer ${token}`;
    } else {
      const userID = await getItem(storageKey.USER_ID_KEY);

      if (userID) {
        param.UserID = userID;
      }
    }

    return {
      headers: {
        ...h,
        ...param,
      },
    };
  });

  const errorLink = onError((error) => {
    console.log(error.graphQLErrors);
    if ((error.graphQLErrors || []).length > 0) {
      const graphQLErrors = error.graphQLErrors || [
        { message: 'エラー発生しました' },
      ];

      const message = graphQLErrors[0].message;
      Alert.alert('エラー', message);
    }
  });

  return new ApolloClient({
    link: errorLink.concat(authLink.concat(createHttpLink({ uri }))),
    cache: cache,
  });
};

export default makeApolloClient;
