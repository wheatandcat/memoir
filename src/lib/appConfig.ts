import firebase from 'lib/system/firebase';

export type AppConfig = {
  maintenance: boolean;
  maintenanceMessage: string;
  maintenancePeriod: string | null;
  supportVersion: string;
};

export const defaultAppConfig = () => ({
  maintenance: false,
  maintenanceMessage: '',
  maintenancePeriod: null,
  supportVersion: '1.0.0',
});

export const getAppConfig = async (
  db: firebase.firestore.Firestore
): Promise<AppConfig> => {
  const qs = await db.collection('appConfig').limit(1).get();

  const records = qs.docs.map((elem) => {
    return elem.data();
  });

  if (!records || !records[0]) {
    return defaultAppConfig();
  }

  return records[0] as AppConfig;
};
