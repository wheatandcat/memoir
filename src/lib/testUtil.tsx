import React from 'react';
import { GraphQLHandler, GraphQLRequest } from 'msw';
import {
  ApolloClient,
  ApolloProvider,
  createHttpLink,
  InMemoryCache,
} from '@apollo/client';
import { render } from '@testing-library/react-native';
import { server } from 'mocks/server';
import fetch from 'cross-fetch';

const link = createHttpLink({
  uri: 'http://localhost:8080/query',
  fetch,
  credentials: 'same-origin',
});

const client = new ApolloClient({
  link,
  uri: 'http://localhost:8080/query',
  cache: new InMemoryCache(),
});

export const testRenderer =
  (children: React.ReactNode) =>
  (responseOverride?: GraphQLHandler<GraphQLRequest<never>>) => {
    if (responseOverride) {
      server.use(responseOverride);
    }
    return render(<ApolloProvider client={client}>{children}</ApolloProvider>);
  };
