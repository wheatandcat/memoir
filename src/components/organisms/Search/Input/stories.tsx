import React from 'react';
import { storiesOf } from '@storybook/react-native';
import InputDate, { Props as InputDateProps } from './InputDate';
import InputUsers, { Props as InputUsersProps } from './InputUsers';
import InputCategory, { Props as InputCategoryProps } from './InputCategory';
import { mockFn } from 'storyBookUtils/index';

const inputDateProps = (): InputDateProps => ({
  startDate: new Date('2021-01-01T00:00:00+09:00'),
  endDate: new Date('2021-01-18T00:00:00+09:00'),
  onChangeStartDate: mockFn('onChangeStartDate'),
  onChangeEndDate: mockFn('onChangeStartDate'),
});
const inputUsers = (): InputUsersProps => ({
  users: [
    {
      id: 'test1',
      image: 'https://placehold.jp/150x150.png',
    },
    {
      id: 'test2',
      image: 'https://placehold.jp/150x150.png',
    },
    {
      id: 'test3',
      image: 'https://placehold.jp/150x150.png',
    },
    {
      id: 'test4',
      image: 'https://placehold.jp/150x150.png',
    },
  ],
  userIDList: [],
  onAdd: mockFn('onAdd'),
  onRemove: mockFn('onRemove'),
});
const inputCategoryProps = (): InputCategoryProps => ({});

storiesOf('organisms/Search/Input', module)
  .add('InputDate', () => <InputDate {...inputDateProps()} />)
  .add('InputUsers', () => <InputUsers {...inputUsers()} />)
  .add('InputCategory', () => <InputCategory {...inputCategoryProps()} />);
