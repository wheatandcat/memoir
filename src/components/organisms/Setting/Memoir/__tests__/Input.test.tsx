import React from 'react';
import { testRenderer } from 'lib/testUtil';
import { screen } from '@testing-library/react-native';
import Input, { Props } from '../Input';

const propsData = (): Props => ({
  dayOfWeek: 0,
  time: new Date('0000-01-01T00:00:00'),
  onChangeDayOfWeek: jest.fn(),
  onChangeTime: jest.fn(),
});

describe('components/organisms/Setting/Memoir/Input.tsx', () => {
  beforeEach(() => {
    testRenderer(<Input {...propsData()} />)();
  });

  it('正常にrenderすること', () => {
    expect(screen.findAllByText('')).toBeTruthy();
  });
});
