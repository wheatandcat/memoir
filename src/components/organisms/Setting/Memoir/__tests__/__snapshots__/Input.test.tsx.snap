// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/Setting/Memoir/Input.tsx 正常にrenderすること 1`] = `
<View
  pt={4}
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={Object {}}
    >
      <RNPickerSelect
        Icon={null}
        InputAccessoryView={null}
        disabled={false}
        doneText="Done"
        fixAndroidTouchableBug={true}
        itemKey={null}
        items={
          Array [
            Object {
              "key": 1,
              "label": "日",
              "value": 1,
            },
            Object {
              "key": 2,
              "label": "月",
              "value": 2,
            },
            Object {
              "key": 3,
              "label": "火",
              "value": 3,
            },
            Object {
              "key": 4,
              "label": "水",
              "value": 4,
            },
            Object {
              "key": 5,
              "label": "木",
              "value": 5,
            },
            Object {
              "key": 6,
              "label": "金",
              "value": 6,
            },
            Object {
              "key": 7,
              "label": "土",
              "value": 7,
            },
          ]
        }
        modalProps={Object {}}
        onClose={null}
        onDonePress={null}
        onDownArrow={null}
        onOpen={null}
        onUpArrow={null}
        onValueChange={[MockFunction]}
        pickerProps={Object {}}
        placeholder={Object {}}
        style={
          Object {
            "inputAndroid": Object {
              "color": "#362740",
              "fontSize": 36,
              "fontWeight": "bold",
              "height": 50,
              "marginTop": 8,
              "paddingTop": 4,
              "width": 40,
            },
            "inputIOS": Object {
              "color": "#362740",
              "fontSize": 36,
              "fontWeight": "bold",
              "height": 50,
              "paddingTop": 16,
              "width": 40,
            },
            "placeholder": Object {
              "color": "transparent",
              "height": 50,
              "width": 40,
            },
          }
        }
        textInputProps={Object {}}
        touchableDoneProps={Object {}}
        touchableWrapperProps={Object {}}
        useNativeAndroidPickerStyle={false}
        value={0}
      />
    </View>
    <View
      pt={4}
      px={2}
    >
      <Text
        fontFamily="RobotoCondensed-Bold"
        variants="body"
      >
        曜日の
      </Text>
    </View>
    <View
      mt={1}
      pt={2}
    >
      <TouchableOpacity>
        <Text
          fontFamily="RobotoCondensed-Bold"
          fontWeight="bold"
          onPress={[Function]}
          style={Object {}}
          variants="large"
        >
          00:00
        </Text>
      </TouchableOpacity>
      <DateTimePickerModal
        backdropStyleIOS={Object {}}
        cancelTextIOS="キャンセル"
        confirmTextIOS="完了"
        date={-000001-12-31T14:41:01.000Z}
        is24Hour={true}
        isVisible={false}
        locale="ja"
        modalPropsIOS={Object {}}
        mode="time"
        onCancel={[Function]}
        onConfirm={[Function]}
        pickerContainerStyleIOS={Object {}}
        pickerStyleIOS={Object {}}
      />
    </View>
    <View
      pt={4}
      px={2}
    >
      <Text
        fontFamily="RobotoCondensed-Bold"
        variants="body"
      >
        から
      </Text>
    </View>
  </View>
</View>
`;
